python/src/main.cpp

```diff
@@ -114,939 +114,857 @@ PYBIND11_MODULE(ncnn, m)
-    .def("depth", (Mat(Mat::*)(int)) & Mat::depth, py::arg("z"))
-    //.def("depth", (const Mat (Mat::*)(int) const) & Mat::depth, py::arg("z"))
-    .def(
-        "row",
-    [](Mat& m, int y) {
-        if (m.elempack != 1)
-        {
-            std::stringstream ss;
-            ss << "get ncnn.Mat row only elempack 1 support now, but given " << m.elempack;
-            pybind11::pybind11_fail(ss.str());
-        }
+        .def(py::init(
+                 [](py::tuple shape, size_t elemsize, int elempack, Allocator* allocator) {
+                     Mat* mat = nullptr;
+                     switch (shape.size())
+                     {
+                     case 1:
+                         mat = new Mat(shape[0].cast<int>(), elemsize, elempack, allocator);
+                         break;
+                     case 2:
+                         mat = new Mat(shape[0].cast<int>(), shape[1].cast<int>(), elemsize, elempack, allocator);
+                         break;
+                     case 3:
+                         mat = new Mat(shape[0].cast<int>(), shape[1].cast<int>(), shape[2].cast<int>(), elemsize, elempack, allocator);
+                         break;
+                     case 4:
+                         mat = new Mat(shape[0].cast<int>(), shape[1].cast<int>(), shape[2].cast<int>(), shape[3].cast<int>(), elemsize, elempack, allocator);
+                         break;
+                     default:
+                         std::stringstream ss;
+                         ss << "shape must be 1, 2, 3 or 4 dims, not " << shape.size();
+                         pybind11::pybind11_fail(ss.str());
+                     }
+                     return mat;
+                 }),
+             py::arg("shape"), py::kw_only(),
+             py::arg("elemsize") = 4, py::arg("elempack") = 1, py::arg("allocator") = nullptr)
+        .def(py::init<int, size_t, int, Allocator*>(),
+             py::arg("w"), py::kw_only(),
+             py::arg("elemsize") = 4, py::arg("elempack") = 1, py::arg("allocator") = nullptr)
+        .def(py::init<int, int, size_t, int, Allocator*>(),
+             py::arg("w"), py::arg("h"), py::kw_only(),
+             py::arg("elemsize") = 4, py::arg("elempack") = 1, py::arg("allocator") = nullptr)
+        .def(py::init<int, int, int, size_t, int, Allocator*>(),
+             py::arg("w"), py::arg("h"), py::arg("c"), py::kw_only(),
+             py::arg("elemsize") = 4, py::arg("elempack") = 1, py::arg("allocator") = nullptr)
+        .def(py::init<int, int, int, int, size_t, int, Allocator*>(),
+             py::arg("w"), py::arg("h"), py::arg("d"), py::arg("c"), py::kw_only(),
+             py::arg("elemsize") = 4, py::arg("elempack") = 1, py::arg("allocator") = nullptr)
+
+        .def(py::init<const Mat&>(), py::arg("m"))
+
+        .def(py::init([](py::buffer const b) {
+                 py::buffer_info info = b.request();
```

Comment:
Only this function is changed and a new function `numpy()` is added. All other changes are formated by `clang-format` in this file.